### PR TITLE
Include CSEK headers in V4 URL signing

### PR DESCRIPTION
**Pull request by jskeet**
(The key and key hash still can't yet be included.)

cc @frankyn

### CODE REVIEW

**Comment by jskeet** [comment]
Hold off on reviewing this - a decision has been taken that will make this a bigger change

**Comment by chrisdunelm** [comment]
OK, ping me when you want it reviewed.

File: apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/UrlSigner.V4Signer.cs [hunk 1]
```diff
@@ -170,7 +170,7 @@ void AddHeaders(SortedDictionary<string, string> canonicalized, IDictionary<stri
                                 continue;
                             }
                             var headerName = pair.Key.ToLowerInvariant();
-                            if (headerName == EncryptionKey.KeyHeader || headerName == EncryptionKey.KeyHashHeader || headerName == EncryptionKey.AlgorithmHeader)
+                            if (headerName == EncryptionKey.KeyHeader || headerName == EncryptionKey.KeyHashHeader)
```
Comment (frankyn) [inline apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/UrlSigner.V4Signer.cs]: I'm expecting this code to also be removed given today's meeting.

File: apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/UrlSignerV4TestData.json [hunk 2]
```diff
@@ -121,14 +121,14 @@
...
+        "expectedUrl": "https://storage.googleapis.com/test-bucket/test-object?X-Goog-Algorithm=GOOG4-RSA-SHA256&X-Goog-Credential=test-iam-credentials%40dummy-project-id.iam.gserviceaccount.com%2F20190201%2Fauto%2Fstorage%2Fgoog4_request&X-Goog-Date=20190201T090000Z&X-Goog-Expires=10&X-Goog-SignedHeaders=host%3Bx-goog-encryption-algorithm&X-Goog-Signature=5cd533eae44467fcc55ba449228a178e1afb39a72c1e01c3b6c8408451169f01c7af634804bcacc9c437dfbc5c1e8ce82bba4d7a1a654734e790ba3bff036cbbb04f564c4dec91717d6f671bd564096d600b4fcc8663b35a520bc15451bcc529edc9a219f8ba7369d0de66da524e899989260f65825ddf2d6539e21c4a88461c01daa1c9e06bc5b8f827d20f712cd44904b4a316f4a4e85d187f80e114929edd9c30e1cf3ff7f04fb01ac7e75f1c7a22e2f4047f73c57a315ec59089243401f00c41391dc108ffc39fe8fc393a71a4fb899f4178fbbb609dfbe33a5d7f69029f233ea8bdc612a2c59b4d620c2a10a5d5222a51039754581d0b111120fa1b88b1"
```
Comment (frankyn) [inline apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/UrlSignerV4TestData.json]: This URL should be updated to include the key-sha256 and key in signedHeaders

**Comment by jskeet** [comment]
Full changes now done, PTAL.

File: apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/UrlSignerV4TestData.json (hunk 2 above)
Comment (jskeet) [reply apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/UrlSignerV4TestData.json]: Now updated.

File: apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/UrlSigner.V4Signer.cs (hunk 1 above)
Comment (jskeet) [reply apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/UrlSigner.V4Signer.cs]: Done.

**Comment by frankyn** [comment]
Thanks @jskeet!

**Pull request merged**